src/plugins/data/public/antlr/shared/utils.ts

```diff
@@ -4,81 +4,85 @@
  */
 
 import { from } from 'rxjs';
-import { distinctUntilChanged, switchMap } from 'rxjs/operators';
+import { distinctUntilChanged, startWith, switchMap } from 'rxjs/operators';
+import { DataSetManager } from '../../query';
 
 export interface IDataSourceRequestHandlerParams {
   dataSourceId: string;
   title: string;
 }
 
+// Function to get raw suggestion data
 export const getRawSuggestionData$ = (
-  connectionsService: any,
-  dataSourceReuqstHandler: ({
+  dataSetManager: DataSetManager,
+  dataSourceRequestHandler: ({
     dataSourceId,
     title,
   }: IDataSourceRequestHandlerParams) => Promise<any>,
-  defaultReuqstHandler: () => any
+  defaultRequestHandler: () => Promise<any>
 ) =>
-  connectionsService.getSelectedConnection$().pipe(
+  dataSetManager.getUpdates$().pipe(
+    startWith(dataSetManager.getDataSet()),
     distinctUntilChanged(),
-    switchMap((connection: any) => {
-      if (connection === undefined) {
-        return from(defaultReuqstHandler());
+    switchMap((dataSet) => {
+      if (!dataSet) {
+        return from(defaultRequestHandler());
       }
-      const dataSourceId = connection?.dataSource?.id;
-      const title = connection?.attributes?.title;
-      return from(dataSourceReuqstHandler({ dataSourceId, title }));
+      const dataSourceId = dataSet?.dataSourceRef?.id;
+      const title = dataSet?.dataSourceRef?.name;
+      return from(dataSourceRequestHandler({ dataSourceId, title }));
```

Comment:
if dataSourceId and title are undefined, would this fail? if not then make `IDataSourceRequestHandlerParams` to define them as optional string?

Reply:
yes, we should make `IDataSourceRequestHandlerParams` optional string